First run. (#1)
chadwcarlson/probot: docs/deployment.md

  ```
  mkdir .platform
  touch .platform/routes.yaml && touch .platform/services.yaml
  touch .platform.app.yaml
  ```

  Unless your Probot app depends on [database storage](https://probot.github.io/docs/persistence/), the `services.yaml` file can be left empty. If a database is needed, visit Platform.sh [Services documentation](https://docs.platform.sh/configuration/services.html) for more detailed information on how to configure the service your app depends on.

  In your [`routes.yaml`](https://docs.platform.sh/configuration/routes.html) file, copy the following:

  ```yaml
  "https://{default}/":
    type: upstream
    upstream: "app:http"

  "https://www.{default}/":
      type: redirect
      to: "https://{default}/"
  ```

  This configuration directs requests to an application called `app`, and sets an additional redirect for requests that do not include the `www` prefix.

  Lastly, in the [`.platform.app.yaml`](https://docs.platform.sh/configuration/app-containers.html) file, include the settings below:

  ```yaml
  name: app

>   type: 'nodejs:12'

Reviewer: maybe put an X and the reference to NodeJS documentation, so we don't need to remember to update this page to each NodeJS upgrade.

Author: Good thinking. Updated.

1. Visit `https://your-deployment.now.sh/api` to invoke the serverless function.

### Platform.sh

> There are two ways to deploy your Probot app to [Platform.sh](https://platform.sh/): [manually](#deploying-an-existing-project) or by modifying an existing [template project](#quickstart-deploying-a-template). Either will work, however the template has been design to set the environment variables retrieved from GitHub during registration automatically, for every branch of your project, and may prove easier for migration in the long run.

Reviewer 2: ```suggestion
There are two ways to deploy your Probot app to [Platform.sh](https://platform.sh/): [manually](#deploying-an-existing-project) or by modifying an existing [template project](#quickstart-deploying-a-template). Either will work, however the template has been designed to set the environment variables retrieved from GitHub during registration automatically, for every branch of your project, and may prove easier for migration in the long run.
```

<a href="https://console.platform.sh/projects/create-project?template=https://raw.githubusercontent.com/platformsh/template-builder/master/templates/probot/.platform.template.yaml&utm_content=probot&utm_source=github&utm_medium=button&utm_campaign=deploy_on_platform">
    <img src="https://platform.sh/images/deploy/lg-blue.svg" alt="Deploy on Platform.sh" width="180px" />
</a>
</p>

The link will create a free trial account on Platform.sh, create a new project to host the app, and then deploy the [Probot template repository](https://github.com/platformsh-templates/probot) on that project.

After that, there are only a few steps left to set up and register the app:

1. Install the Platform.sh CLI

  Setting relevant environment variables and retrieving information about the Probot project will be much easier once you install the [Platform.sh CLI](https://docs.platform.sh/development/cli.html). Run the following command:

  ```
  curl -sS https://platform.sh/cli/installer | php
  ```

  You can find the system requirements and more information in the [installation instructions on GitHub](https://github.com/platformsh/platformsh-cli/blob/master/README.md#installation).

> 1. Follow the remaining steps in the README

Reviewer 2: Is the rest of the text in this section necessary?  The README in the template already covers it.  Just link to that rather than duplicating it.

1. Add Platform.sh configuration files

  Each project on Platform.sh requires at least [three configuration files](https://docs.platform.sh/overview/structure.html) to describe and application. From your project root, run the following commands to create them:

  ```
  mkdir .platform
  touch .platform/routes.yaml && touch .platform/services.yaml
  touch .platform.app.yaml
  ```

  Unless your Probot app depends on [database storage](https://probot.github.io/docs/persistence/), the `services.yaml` file can be left empty. If a database is needed, visit the Platform.sh [Services documentation](https://docs.platform.sh/configuration/services.html) for more detailed information on how to configure the service your app depends on.

  In your [`routes.yaml`](https://docs.platform.sh/configuration/routes.html) file, copy the following:

  ```yaml
  "https://{default}/":
    type: upstream
    upstream: "app:http"

  "https://www.{default}/":
      type: redirect
      to: "https://{default}/"
  ```

>   This configuration directs requests to an application container called `app`, and sets an additional redirect for requests that do not include the `www` prefix.

Reviewer 2: It's actually backward from this description.  The redirect is added to URLs that DO have a www.

  Lastly, in the [`.platform.app.yaml`](https://docs.platform.sh/configuration/app-containers.html) file, include the settings below:

  ```yaml
  name: app

>   type: 'nodejs:X'

Reviewer 2: ```suggestion
  type: 'nodejs:12'
```

Let's go ahead and make this fully copy-pasteable.  We can still document that users can change it but don't make them have to ask that question out of the gate.

  variables:
      env:
          NODE_ENV: development

  disk: 256

  web:
    commands:
      start: "npm start"
  ```

  This file defines your Probot app to run in a Node.js application container called `app`, with 256Mb of persistent storage. Feel free to modify the `web.commands.start` to match the start command of your app.

>   If you would like to run the Probot app on Node.js 12, simply modify the `type` key to read `'nodejs:12'` instead of the `X` placeholder shown above. You can set the application container to use a number of Node.js versions, so visit the [documentation](https://docs.platform.sh/languages/nodejs.html#supported-versions) for a full list of those supported. 

Reviewer 2: ```suggestion
  If you would like to run the Probot app on a different version of Node.js, modify the `type` key to read `'nodejs:X'` where X is any of the supported versions listed in the Platform.sh [documentation](https://docs.platform.sh/languages/nodejs.html#supported-versions). 
```
